src/utils/accessions.ts

```diff
@@ -0,0 +1,41 @@
+export const studyAccessionPattern = /^MGYS[0-9]{8}$/;
+
+export const analysisAccessionPattern = /^MGYA[0-9]{8}$/;
+
+export const genomeAccessionPattern = /^MGYG[0-9]{9}$/;
+
+export const mgnifyAccessionPattern = /^MGY[AGSP]{1}[0-9]{1,12}$/;
+
+type DetailOrSearchURLType = {
+  isAccessionLike: boolean;
+  nextURL: string;
+};
+
+export const getDetailOrSearchURLForQuery = (
```

Comment:
@SandyRogers perhaps we can reduce nesting here:`const nextUrlsBasedOnAccessionPatterns = {
'/^MGY[AGSP]{1}[0-9]{1,12}$/' : '/studies/',
'/^MGYG[0-9]{9}$/': '/analyses', 
}

//then the function can just have 1 return statement:

return {
isAccessionLike: true,
nextURL: nextUrlsBasedOnAccessionPatterns[mgnifyAccessionPattern]
}

`